src/main/java/maxsuperman/addons/roller/modules/VillagerRoller.java

```diff
@@ -715,13 +721,27 @@ private void onTick(TickEvent.Pre event) {
                 placeFailed("Lectern not found in hotbar");
                 return;
             }
-            if (!BlockUtils.canPlace(rollingBlockPos, true)) {
-                placeFailed("Can't place lectern");
-                return;
-            }
-            if (!BlockUtils.place(rollingBlockPos, item, headRotateOnPlace.get(), 5)) {
-                placeFailed("Failed to place lectern");
+            if (baritonePlacing.get()) {
+                int x = rollingBlockPos.getX();
+                int y = rollingBlockPos.getY();
+                int z = rollingBlockPos.getZ();
+                // Use Baritone to place the lectern block
+                BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("sel clear");
+                BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("sel 1 "+x+" "+y+" "+z);
+                BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("sel 2 "+x+" "+y+" "+z);
+                BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("sel f minecraft:lectern");
+                BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("sel clear");
```

Comment:
`GoalPlace` from `baritone.process` should be used (through `ICustomGoalProcess`)